gobblin-cluster/src/main/java/org/apache/gobblin/cluster/HelixUtils.java

```diff
@@ -145,6 +146,31 @@ static void waitJobInitialization(
     log.info("Work flow {} initialized", workFlowName);
   }
 
+  public static boolean deleteTaskFromHelixJob(String workFlowName,
+      String jobName, String taskID, TaskDriver helixTaskDriver) {
+    try {
+      log.info(String.format("try to delete task %s from workflow %s, job %s", taskID, workFlowName, jobName));
+      helixTaskDriver.deleteTask(workFlowName, jobName, taskID);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    return !helixTaskDriver.getJobConfig(TaskUtil.getNamespacedJobName(workFlowName, jobName)).getMapConfigs().containsKey(taskID);
```

Comment:
Do you know what kind of consistency guarantee offered by deleteTask and addTask from helix ? Or rather, what's the intention of read-after-write pattern here ?

Reply:
oh that's a good question, I haven't check that part, just follow the instruction on the doc that when error happens it's caller's responsibility to check whether the update successful. Let me only add this part when exception happens to not rely on the read/write consistency in normal case